lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -421,7 +416,7 @@ void ThreadManager::Impl::removeWorker(size_t value) {
   {
     Synchronized s(workerMonitor_);
 
-    while (workerCount_ != workerMaxCount_) {
+    while (workerCount_ > goalCount) {
```

Comment:
Old line 399 modifies it under the monitor ``monitor_`` and then old line 424 acquired it under a different monitor ``workerMonitor_``.  One cannot use different monitors to control access to the same variable and expect any synchronization.  This is why I squirrel away the new value into goalCount and use it later on.